### PR TITLE
fix #39 support falkordb:// protocol

### LINKED ISSUE #39: add support for falkordb:// connection URL protocol

### PR DESCRIPTION
**Pull request by gkorland**
## Summary by CodeRabbit

- **Bug Fixes**
	- Updated the database connection protocol in the app to enhance compatibility and stability.

### CODE REVIEW

File: src/falkordb.ts [hunk 1]
```diff
@@ -103,6 +103,12 @@ export default class FalkorDB extends EventEmitter {
     static async connect(options?: FalkorDBOptions) {
         const redisOption = (options ?? {}) as RedisClientOptions<{ falkordb: typeof commands }, RedisFunctions, RedisScripts>;
 
+        // If the URL is provided, and the protocl is falkordb replace it with redis for the underline redis client
+        // e.g. falkordb://localhost:6379 -> redis://localhost:6379
+        if (redisOption.url && redisOption.url.startsWith('falkordb')) {
```
Comment (AviAvni) [inline src/falkordb.ts:108]: take a look at https://github.com/FalkorDB/falkordb-py/blob/6613929b980dbac426f98a7e9bc2e7538d9e80ec/falkordb/falkordb.py#L125 it is falkor not falkordb

Comment (gkorland) [reply src/falkordb.ts:108]: so falkor:// and not falkordb://?

File: src/falkordb.ts [hunk 2]
```diff
@@ -103,6 +103,12 @@
...
+            redisOption.url = redisOption.url.replace('falkordb', 'redis');
```
[the remark this answers is not in the document]
Comment (gkorland) [reply src/falkordb.ts]: user:pass@falkordb://localhost:6379 is not a valid URL

[the remark this answers is not in the document]
Comment (gkorland) [reply src/falkordb.ts]: the redis supported format URL for authentication is 
`redis://user:pass@localhost:6379`
Same for Falkor
`falkor://user:pass@localhost:6379`

**Review by AviAvni — APPROVED** [review]

**Pull request merged**